applications/StructuralMechanicsApplication/custom_utilities/constitutive_law_utilities.cpp

```diff
@@ -336,15 +334,14 @@ void ConstitutiveLawUtilities<TVoigtSize>::CalculateElasticMatrixPlaneStrain(Mat
 /***********************************************************************************/
 /***********************************************************************************/
 template<SizeType TVoigtSize>
-void ConstitutiveLawUtilities<TVoigtSize>::CalculateElasticMatrix(MatrixType& rC, ConstitutiveLaw::Parameters& rValues)
+void ConstitutiveLawUtilities<TVoigtSize>::CalculateElasticMatrix(
+    MatrixType& rC,
+    const double E,
+    const double NU)
 {
-    const Properties& r_material_properties = rValues.GetMaterialProperties();
-    const double E  = r_material_properties[YOUNG_MODULUS];
-    const double NU = r_material_properties[POISSON_RATIO];
-
     if (rC.size1() != VoigtSize)
         rC.resize(VoigtSize, VoigtSize, false);
-    noalias(rC) = ZeroMatrix(VoigtSize, VoigtSize);
+    rC.clear();
```

Comment:
are we sure this is the same?

Reply:
@loumalouomega

Reply:
Indeed it is better. `ZeroMatrix` allocates a new one while `clear` just wipes everything inside.